source/dub/compilers/compiler.d

```diff
@@ -200,20 +204,17 @@ interface Compiler {
 			build_platform.compilerVersion = ver;
 		}
 
-		// Skip the following check for LDC, emitting a warning if the specified `-arch`
-		// cmdline option does not lead to the same string being found among
-		// `build_platform.architecture`, as it's brittle and doesn't work with triples.
-		if (build_platform.compiler != "ldc") {
-			if (arch_override.length && !build_platform.architecture.canFind(arch_override) &&
-				!(build_platform.compiler == "dmd" && arch_override.among("x86_omf", "x86_mscoff")) // Will be fixed in determinePlatform
-			) {
-				logWarn(`Failed to apply the selected architecture %s. Got %s.`,
-					arch_override, build_platform.architecture);
-			}
-		}
-
 		return build_platform;
 	}
+
+	/// Adds the given flags to the build settings if desired, otherwise informs the user
+	final void maybeAddArchFlags(bool keep_arch, ref BuildSettings settings, string[] arch_flags, string arch_override) {
```

Comment:
Maybe this can become `protected static` instead of `public final`?

Reply:
Or move it outside the `class`:
```D
package void maybeAddArchFlags(ref BuildSettings settings, bool keep_arch, string[] arch_flags, string arch_override)
```

Reordering args to make it UFCS-friendly.